app.js

```diff
@@ -1,11 +1,33 @@
 const express = require("express");
-const { getTopics, getArticles } = require("./controllers/index.js");
+const {
+  getTopics,
+  getArticles,
+  getArticleById,
+} = require("./controllers/index.js");
 
 const app = express();
 
 app.use(express.json());
 
 app.get("/api/topics", getTopics);
 app.get("/api/articles", getArticles);
+app.get("/api/articles/:article_id", getArticleById);
+
+app.use((err, req, res, next) => {
+  console.log(err);
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Invalid query datatype" });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+});
```

Comment:
Nice error handlers. Remove the console.log though when committing.
Also, you'll want to get a 500s error handler in place where you DO keep a console.log